src/plugins/fs/vfat.c

```diff
@@ -274,7 +282,14 @@ gboolean bd_fs_vfat_set_label (const gchar *device, const gchar *label, GError *
             args[2] = "--reset";
     }
 
-    return bd_utils_exec_and_report_error (args, NULL, error);
+    /* forcefully convert the label uppercase */
+    if (!new_vfat && label)
```

Comment:
Why only for dosfstools < 4.2? You convert it for both the old and new version in `bd_fs_vfat_mkfs_options`.

Reply:
That's a logical conundrum. `new_vfat` is only set when dosfstools >= 4.2 AND the label is empty.

Reply:
Oh, I see. But I think adding a new `is_empty`  boolean would make this much easier to read.

Reply:
Attempt #2, please check.